src/plonk/keygen.rs

```diff
@@ -218,21 +222,27 @@ where
     let mut cs = ConstraintSystem::default();
     let config = ConcreteCircuit::configure(&mut cs);
 
+    let cs = cs;
+    // There needs to be enough room for at least one row.
+    assert!(
+        cs.blinding_factors() // m blinding factors
+        + 1 // for l_{-(m + 1)}
+        + 1 // for l_0
+        + 1 // for at least one row
+        <= (params.n as usize)
+    );
```

Comment:
Duplicate code. Consider adding a `cs.minimum_rows()` method. (This could also be used in MockProver where `cs.blinding_factors() + 3` [is hard-coded](https://github.com/zcash/halo2/pull/316/files?diff=unified&w=1#diff-da17e7db207e483590bb92572e40b583f57b8737739a902946b46bc0d59b0465R411).)